shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteDetailsStopList.kt

```diff
@@ -70,12 +82,34 @@ data class RouteDetailsStopList(val stops: List<Entry>) {
                         val stop =
                             globalData.getStop(stopId)?.resolveParent(globalData)
                                 ?: return@mapNotNull null
+                        val patterns = globalData.getPatternsFor(stopId, routeId)
                         val transferRoutes =
                             TripDetailsStopList.getTransferRoutes(stopId, routeId, globalData)
-                        Entry(stop, transferRoutes)
+                        Entry(stop, patterns, transferRoutes)
                     }
 
-                RouteDetailsStopList(stops)
+                val segments = splitIntoSegments(stops)
+
+                RouteDetailsStopList(segments)
             }
+
+        fun splitIntoSegments(entries: List<Entry>): List<Segment> {
+            val (accSegments, lastWipSegment) =
+                entries.fold(Pair<List<Segment>, Segment?>(listOf(), null)) {
```

Comment:
I think just using a `MutableList<Segment>` and a `for` loop might be more legible here than `fold` with a `Pair` accumulator.